src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -94,6 +99,45 @@ private Task OnProjectFactoryCompletedAsync()
 
             // Update project context and subscriptions.
             await UpdateProjectContextAndSubscriptionsAsync().ConfigureAwait(false);
+            LogLanguageServiceHostInitializeStop();
+        }
+
+        private static volatile bool s_isFirstStartEvent = true;
+        private static volatile int s_numberOfTimesInitialized = 0;
+        private static object s_telemetryLock = new object();
+
+        /// <summary>
+        /// Only posts an event the first time this method is called in the VS process
+        /// Also updates a global count so we know how many project hosts have been initialized
+        /// </summary>
+        private void LogLanguageServiceHostInitializeStart()
+        {
+            if (s_isFirstStartEvent)
+            {
+                lock (s_telemetryLock)
+                {
+                    if (s_isFirstStartEvent)
+                    {
+                        s_isFirstStartEvent = false;
+                        _telemetryService.PostEvent("LanguageServiceHost/Start");
+                    }
+                }
+            }
+
+            Interlocked.Increment(ref s_numberOfTimesInitialized);
+        }
+
+        /// <summary>
+        /// Decrements the global count and reports that all hosts have been initialized
+        /// if the count is zero.
+        /// </summary>
+        private void LogLanguageServiceHostInitializeStop()
+        {
+            Interlocked.Decrement(ref s_numberOfTimesInitialized);
+            if (s_numberOfTimesInitialized == 0)
```

Comment:
You should be checking the result of `Interlocked.Decrement`, not the field.  Otherwise, it might have already changed again by the time you compare it.